udata/core/topic/apiv2.py

```diff
@@ -0,0 +1,258 @@
+    'tags': fields.List(
+        fields.String, description='Some keywords to help in search', required=True),
+
+    'datasets': fields.Raw(attribute=lambda o: {
+        'rel': 'subsection',
+        'href': url_for('apiv2.topic_datasets', topic=o.id, page=1,
+                        page_size=DEFAULT_PAGE_SIZE, _external=True),
+        'type': 'GET',
+        'total': len(o.datasets)
+        }, description='Link to the topic datasets'),
+    'reuses': fields.Raw(attribute=lambda o: {
+        'rel': 'subsection',
+        'href': url_for('apiv2.topic_reuses', topic=o.id, page=1,
+                        page_size=DEFAULT_PAGE_SIZE, _external=True),
+        'type': 'GET',
+        'total': len(o.reuses)
+        }, description='Link to the topic reuses'),
+    'featured': fields.Boolean(description='Is the topic featured'),
+    'private': fields.Boolean(description='Is the topic private'),
+    'created_at': fields.ISODateTime(
+        description='The topic creation date', readonly=True),
+    'organization': fields.Nested(
+        org_ref_fields, allow_null=True,
+        description='The publishing organization', readonly=True),
+    'owner': fields.Nested(
+        user_ref_fields, description='The owner user', readonly=True,
+        allow_null=True),
+    'uri': fields.UrlFor(
+        'api.topic', lambda o: {'topic': o},
+        description='The topic API URI', readonly=True),
+    'page': fields.UrlFor(
+        'topics.display', lambda o: {'topic': o},
+        description='The topic page URL', readonly=True, fallback_endpoint='api.topic'),
+    'extras': fields.Raw(description='Extras attributes as key-value pairs'),
+})
+
+topic_page_fields = apiv2.model('TopicPage', fields.pager(topic_fields))
+
+
+@ns.route('/', endpoint='topics_list')
+class TopicsAPI(API):
+    @apiv2.expect(topic_parser.parser)
+    @apiv2.marshal_with(topic_page_fields)
+    def get(self):
+        '''List all topics'''
+        args = topic_parser.parse()
+        topics = Topic.objects()
+        topics = topic_parser.parse_filters(topics, args)
+        sort = args['sort'] or ('$text_score' if args['q'] else None) or DEFAULT_SORTING
+        return (topics.order_by(sort)
+                .paginate(args['page'], args['page_size']))
+
+
+@ns.route('/<topic:topic>/', endpoint='topic', doc=common_doc)
+@apiv2.response(404, 'Topic not found')
+class TopicAPI(API):
+    @apiv2.doc('get_topic')
+    @apiv2.marshal_with(topic_fields)
+    def get(self, topic):
+        '''Get a given topic'''
+        return topic
+
+
+topic_add_items_fields = apiv2.model('TopicItemsAdd', {
+    'id': fields.String(description='Id of the item to add', required=True),
+}, location='json')
+
+
+@ns.route('/<topic:topic>/datasets/', endpoint='topic_datasets', doc=common_doc)
+class TopicDatasetsAPI(API):
+    @apiv2.doc('topic_datasets')
+    @apiv2.expect(generic_parser)
+    @apiv2.marshal_with(dataset_page_fields)
+    def get(self, topic):
```

Comment:
Seems to work well!
But I'm scared that some users *will probably* want to add some filters or sort, redeveloping a more complex dataset list endpoint.

Could we somehow share some code with apiv1 `datasets` list endpoint by adding a topic filter? Or even imagine a redirect?

Reply:
About adding filter: maybe simpler to keep it as is (2 lines of code, generic parser) and mutualize if and when more filters are needed?

About redirect: not sure how it could work 🤔 Won't be to v1 anyway.

Reply:
I'm weak, I used the `DatasetParser` from api v1 in the latest commit ;-)

We should use the topic arg when it's available from #2915.

Only drawback I see is mixing v1 and v2. I don't think this needs dedicated testing, the parser should already be tested elsewhere.

Reply:
Thank you! I've merged https://github.com/opendatateam/udata/pull/2915.
I think mixing both for now is not much of a problem, let's reuse what works :)

Reply:
🤝